dev/update_version.py

```diff
@@ -0,0 +1,127 @@
+        patch_version += 1
+    elif increment == "minor":
+        minor += 1
+        patch_version = 0
+    elif increment == "major":
+        major += 1
+        minor = 0
+        patch_version = 0
+    else:
+        raise ValueError(
+            "Invalid increment type. Must be 'major', 'minor', or 'patch'."
+        )
+    return f"{major}.{minor}.{patch_version}"
+
+
+def _update_versions(file_patterns, replace_strings, new_version, check):
+    """Update the version strings in the specified files."""
+    wrong = False
+    for pattern in file_patterns:
+        files = list(Path(__file__).parents[1].glob(pattern))
+        for file_path in files:
+            if not file_path.is_file():
+                continue
+            content = file_path.read_text()
+            original_content = content
+            for s in replace_strings:
+                # Construct regex pattern to match any version number in the string
+                escaped_s = re.escape(s).replace(r"\{version\}", r"(\d+\.\d+\.\d+)")
+                regex_pattern = re.compile(escaped_s)
+                content = regex_pattern.sub(s.format(version=new_version), content)
+            if content != original_content:
+                wrong = True
+                if check:
+                    print(f"{file_path} would be updated")
+                else:
+                    file_path.write_text(content)
+                    print(f"Updated {file_path}")
+
+    if wrong and check:
+        sys.exit("Some version haven't been updated.")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="Utility used to bump the version of the package."
+    )
+    parser.add_argument("current_version", help="Current version of the package.")
+    parser.add_argument(
+        "--check", action="store_true", help="Fails if any file would be modified."
+    )
+    parser.add_argument(
+        "--examples", action="store_true", help="Also modify flwr version in examples."
+    )
+
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument(
+        "--patch", action="store_true", help="Increment the patch version."
+    )
+    group.add_argument(
+        "--major", action="store_true", help="Increment the major version."
+    )
+    args = parser.parse_args()
+
+    curr_version = args.current_version
+
+    # Determine the type of version increment
+    if args.major:
+        increment = "major"
+    elif args.patch:
+        increment = "patch"
+    else:
+        increment = "minor"
+
+    next_version = _get_next_version(curr_version, increment)
+
+    # Update files with next version
+    for file_pattern, strings in REPLACE_NEXT_VERSION.items():
+        _update_versions([file_pattern], strings, next_version, args.check)
+
+    # Update files with current version
+    for file_pattern, strings in REPLACE_CURR_VERSION.items():
+        _update_versions([file_pattern], strings, curr_version, args.check)
```

Comment:
@Robert-Steiner I see what you mean, I have updated the script to reflect this. I just modified it so if you run `python update_version.py 1.11.1 --patch`, `REPLACE_NEXT_VERSION` files would be updated to the next minor version, i.e `1.12.0` and the `REPLACE_CURR_VERSION` files to `1.11.2` (as `--patch` was provided). Does that make sense?

Reply:
@Robert-Steiner Also I think that now, it only makes sense to infer the current version instead of having to provide it through the CLI, i.e the ideal DX would just be `python update_version.py --patch`. But which file should we infer it from ? The `stable_flwr_version` from `doc/source/conf.py`?

Reply:
@Robert-Steiner Done!